include/h2o.h

```diff
@@ -2135,16 +2138,28 @@ static inline h2o_iovec_t h2o_doublebuffer_prepare(h2o_doublebuffer_t *db, h2o_b
         db->buf = *receiving;
         *receiving = t;
     }
-    if ((db->bytes_inflight = db->buf->size) > max_bytes)
-        db->bytes_inflight = max_bytes;
-    return h2o_iovec_init(db->buf->bytes, db->bytes_inflight);
+    if ((db->_bytes_inflight = db->buf->size) > max_bytes)
+        db->_bytes_inflight = max_bytes;
+    if (db->_bytes_inflight > 0)
```

Comment:
I do not think that we need a test here.

The value is guaranteed to become non-zero, since we return `{NULL, 0}` earlier when both `db->buf->size` and `(*receiving)->size` are zero.

Reply:
How about the case `max_bytes == 0` ? If it's not allowed by convention, we should add an assertion?

Reply:
> How about the case `max_bytes == 0` ? If it's not allowed by convention, we should add an assertion?

I would argue that it should not be allowed, since `max_bytes` is a parameter that is used to restrict the amount of data passed to `h2o_send`, and since in the general case passing a zero-sized buffer to `h2o_send` needs to be prohibited. Otherwise, there is a chance of entering an infinite loop.

Whether we should add an assertion is another question. Considering the fact that we pass `req->preferred_chunk_size` as `max_bytes`, it might make sense to check the places that we set `req->preferred_chunk_size` first.

Reply:
fixed at https://github.com/h2o/h2o/pull/1508/commits/4eb62ffe8bcefb6edba761e33cfd791352a09e4b